src/app/settings/premium.component.ts

```diff
@@ -76,6 +78,7 @@ export class PremiumComponent implements OnInit {
                     return this.finalizePremium();
                 });
             } else {
+                await this.taxInfoComponent.submitTaxInfo();
```

Comment:
I would prefer we include the tax info in a single request like we do with org create, with postPremium here. Looking at your server code, I don't see how this would even properly work for a User who does not have a Stripe customer record yet. That gets created in postPremium. I think that the same issue exists in other places where you are calling two separate API calls, like add Credit.

Reply:
Yeah, I had almost gone that route, but there was a lot of back and forth trying to minimize duplicative code and reusable components. I'll refactor the server and then ensure these follow suit to append/tack-on the tax info to any calls that would normally use it; while keeping the independent "save" there for organizations from their settings page (outside of billing).

Reply:
**// TODO/:**

(each includes UI, jslib and server changes)
1. `add-credit.component.ts` > Need to refactor server method to accept tax info, append tax-info to add-credit collective data/JSON
2. `adjust-payment.component.ts` > Add tax info to the `PaymentRequest` type (similar to `organization-plans.component.ts`
3. `premium.component.ts` > Add tax info to `FormData` to submit for `postPremium(FormData)`

Reply:
For add credit, that may need to stay a separate step since I don't think we make a call to our server during that action. Credit is added asynchronously via webhook.

Reply:
Yeah, I'm seeing that... I'll check to see if those services enable state transfer or a custom blob/field that I can encode and pass that info across and then use from the web-hook. But otherwise have to kind of rely on them already having a Stripe customer id, otherwise it would fail/be ignored. 

This _may_ not be necessary at all since I'm assuming we're not realizing that revenue (and therefore no tax) when a credit is added and **only** when that credit is applied to an invoice for premium subscription services would we count that ARR/revenue. In that same light, the modification planned for `premium.component.ts` will handle requiring tax information anyway so we may not need this at all when adding account credit since we will always have the billing tax info when actually invoicing (either the first time or from subsequent subscription renewal invoices in Stripe).